pkg/api/controller.go

```diff
@@ -3854,6 +3854,78 @@ func (c *Controller) PostStatsEvents(w http.ResponseWriter, r *http.Request, bod
 	writeResponse(w, r, http.StatusNoContent, nil)
 }
 
+func (c *Controller) OtfDiff(w http.ResponseWriter, r *http.Request, repository string, leftRef string, rightRef string, params OtfDiffParams) {
+	ctx := r.Context()
+	user, err := auth.GetUser(ctx)
+	if err != nil {
+		writeError(w, r, http.StatusUnauthorized, ErrAuthenticatingRequest)
+		return
+	}
+	c.LogAction(ctx, "table_format_diff", r, repository, rightRef, leftRef)
+	credentials, _, err := c.Auth.ListUserCredentials(ctx, user.Username, nil)
+	if c.handleAPIError(ctx, w, r, err) {
+		return
+	}
+	if len(credentials) == 0 {
+		writeError(w, r, http.StatusPreconditionFailed, "no programmatic credentials")
+		return
+	}
+
+	// TODO: pass correct base ref
+	tp := buildS3TablePaths(repository, leftRef, rightRef, "", params.TablePath)
+
+	diffParams := diff.Params{
+		TablePaths: tp,
+		S3Creds: diff.S3Creds{
+			Key:      credentials[0].AccessKeyID,
+			Secret:   credentials[0].SecretAccessKey,
+			Endpoint: fmt.Sprintf("https://%s", c.Config.ListenAddress),
+		},
+	}
+
+	entries, err := c.otfDiffService.RunDiff(ctx, params.Type, diffParams)
+	if err != nil {
+		c.Logger.Error(err)
+		if errors.Is(err, diff.ErrOTFNotFound) {
```

Comment:
does the error include the ref that was not found?

Reply:
no

Reply:
We wanted to add it to understand if a table was removed or added.